### DISCUSSION TITLE
Standalone generating of changelog

### DISCUSSION THREAD

**Discussion by schmkr** [renovatebot/renovate#21483] [Suggest an Idea]
### Type of discussion.

I'm proposing an idea

### Tell us more.

We use Renovate a lot in our projects and one of the things I like most about it are the changelogs in PR descriptions. It allows us to understand what has changed in the changed package(s) and asses if it could cause any issues (that might not be caught by tests or builds in a pipeline).

However, sometimes we still have to manually update some package. In our case, using [Nx](https://nx.dev), whenever there is an update (especially a major or minor update) we also need to run _migrations_ and the side effects of those migrations can be that other packages also get updated (e.g. Angular). Those kind of updates we then do manually (often times, after the migrations we also need to make changes to address things that were broken as a result of the updates).

It would be really awesome if we could generate the same kind of changelog markdown using a standalone command (e.g. `renovate-changelog`) for a give package, set of packages or a diff between 2 commits.

Assuming all the code for generating such changelog is already in the repo, who feasible would it be to create such a tool/sub-command?

**Comment by HonkingGoose** [comment]
> It would be really awesome if we could generate the same kind of changelog markdown using a standalone command (e.g. `renovate-changelog`) for a given package, set of packages or a diff between 2 commits.

It's not completely what you want, but have you tried the [Octoclairvoyant](https://octoclairvoyant.vercel.app/) website? You can compare between tagged versions of packages on public GitHub repositories. It can't compare commits, or give you a diff though.

> Assuming all the code for generating such changelog is already in the repo, who feasible would it be to create such a tool/sub-command?

We probably already have the changelog generating code in our repository. I don't know how much work it would be to create that tool/sub-command and maintain it. I'll let the maintainers decide if this is something they want to invest time in. Also maintainers: let me know if you want an issue for this idea. 😉

**Comment by rarkins** [comment]
I don't mind if someone wants to contribute a PR to refactor/modularity our existing changelog functions so that they can be used elsewhere

**Reply by HonkingGoose** [reply]
Opened an issue to track this:

- #21602
